Sources/SwiftIfConfig/StaticBuildConfiguration.swift

```diff
@@ -0,0 +1,402 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2025 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#if compiler(>=6)
+@_spi(ExperimentalLanguageFeatures) public import SwiftParser
+public import SwiftSyntax
+#else
+@_spi(ExperimentalLanguageFeatures) import SwiftParser
+import SwiftSyntax
+#endif
+
+/// A statically-determined build configuration that can be used with any
+/// API that requires a build configuration. Static build configurations can
+/// be (de-)serialized via Codable.
+///
+/// Static build configurations can not be used for canImport checks, because
+/// such checks require deeper integration with the compiler itself.
+public struct StaticBuildConfiguration: Codable {
+  public init(
+    customConditions: Set<String> = [],
+    features: Set<String> = [],
+    attributes: Set<String> = [],
+    targetOSNames: Set<String> = [],
+    targetArchitectures: Set<String> = [],
+    targetEnvironments: Set<String> = [],
+    targetRuntimes: Set<String> = [],
+    targetPointerAuthenticationSchemes: Set<String> = [],
+    targetObjectFileFormats: Set<String> = [],
+    targetPointerBitWidth: Int = 64,
+    targetAtomicBitWidths: [Int] = [],
+    endianness: Endianness = .little,
+    languageVersion: VersionTuple,
+    compilerVersion: VersionTuple
+  ) {
+    self.customConditions = customConditions
+    self.features = features
+    self.attributes = attributes
+    self.targetOSNames = targetOSNames
+    self.targetArchitectures = targetArchitectures
+    self.targetEnvironments = targetEnvironments
+    self.targetRuntimes = targetRuntimes
+    self.targetPointerAuthenticationSchemes = targetPointerAuthenticationSchemes
+    self.targetObjectFileFormats = targetObjectFileFormats
+    self.targetPointerBitWidth = targetPointerBitWidth
+    self.targetAtomicBitWidths = targetAtomicBitWidths
+    self.endianness = endianness
+    self.languageVersion = languageVersion
+    self.compilerVersion = compilerVersion
+  }
+
+  /// The set of custom conditions that are present and can be used with `#if`.
+  ///
+  /// Custom build conditions can be set by the `-D` command line option to
+  /// the Swift compiler. For example, `-DDEBUG` sets the custom condition
+  /// named `DEBUG`, which could be checked with, e.g.,
+  ///
+  /// ```swift
+  /// #if DEBUG
+  /// // ...
+  /// #endif
+  /// ```
+  public var customConditions: Set<String> = []
```

Comment:
The compiler calls these *conditional compilation flags*. Should the variable be called `conditionalCompilationFlags` or similar to match the terms?

> `-D <value>              Marks a conditional compilation flag as true`

Reply:
That’s fair. Maybe an unfortunate choice of terminology on `BuildConfiguration` then, but that ship has sailed. 🤷🏽